### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by jburel**
Migration of training scripts

To discuss the 2 "very" small exercices
cc @pwalczysko @will-moore 

Ref: https://trello.com/c/XVzyrRqg/2-analysis-workshop

### CODE REVIEW

**Comment by jburel** [comment]
@pwalczysko I have migrated and converted as discussed exercises into example

File: practical/python/scipy_gaussian_filter.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+
+#   You should have received a copy of the GNU General Public License along
+#   with this program; if not, write to the Free Software Foundation, Inc.,
+#   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+# ------------------------------------------------------------------------------
+
+"""
+Run Gaussian Filter from Scipy on Project/Dataset/Image/Screen/Plate/Well.
+"""
```
Comment (will-moore) [inline practical/python/scipy_gaussian_filter.py]: Maybe add here this is a server-side OMERO.script (unless that info is elsewhere in some README?)

File: practical/python/simple_frap.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+
+#   You should have received a copy of the GNU General Public License along
+#   with this program; if not, write to the Free Software Foundation, Inc.,
+#   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+# ------------------------------------------------------------------------------
+
+"""Simple FRAP plots from Rectangles on images."""
```
Comment (will-moore) [inline practical/python/simple_frap.py]: Same for this script. I guess if we have a mix of simple python client scripts and OMERO.scripting service scripts they should be distinguished. e.g. different directories etc.

Comment (jburel) [reply practical/python/simple_frap.py]: We could add ``python/server_scripts/*``

File: practical/javascript/figure_labels_from_mapannotation.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+// Adds labels to image panels in OMERO.figure
+// using Key Value pairs (map annotations) on the images.
+//
+// To use, open the OMERO.figure file in your browser.
+// Select the image panels you want to add labels to.
+// Open devtools in Chrome / Firefox and select Console tab.
+// Copy the code below into the console.
+// This will load map annotations JSON for each selected image and create labels.
+// Can observe under the Network tab in devtools to see JSON loaded.
```
Comment (pwalczysko) [inline practical/javascript/figure_labels_from_mapannotation.js]: That is a bit of a weird workflow.

File: practical/javascript/figure_movie.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+// Creates a Movie layout for selected images
+// with rows of panels showing time-points incremented by tIncrement
+// and wrapped into multiple rows according to columnCount.
```
Comment (pwalczysko) [inline practical/javascript/figure_movie.js]: Insert empty lines to be in sync with other scripts here.

File: practical/java/OmeroBackgroundSubtractBatch.java [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        buffer.append(groupId);
+        buffer.append("\niid=");
+        buffer.append(imageId);
+        buffer.append("]");
+        buffer.append(" windowless=true ");
+        IJ.runPlugIn("loci.plugins.LociImporter", buffer.toString());
+    }
+
+    /**
+     * Connects to OMERO and returns a agteway instance allowing to interact
```
Comment (will-moore) [inline practical/java/OmeroBackgroundSubtractBatch.java]: typo

File: practical/jython/annotate_dataset.jy [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+#  GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+#------------------------------------------------------------------------------
+
+# This Jython script shows how to create and annotate a dataset.
+# The purpose of the script is to be used in the Scripting Dialog
```
Comment (pwalczysko) [inline practical/jython/annotate_dataset.jy]: ``The purpose of the script is to be used in the Scripting Dialog...`` -> ``Use this script in the Scripting Dialog...``

File: practical/jython/background_subtract_batch.jy [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+#  GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+#------------------------------------------------------------------------------
+
+# This Jython script uses ImageJ to Subtract Background.
+# The purpose of the script is to be used in the Scripting Dialog
```
Comment (pwalczysko) [inline practical/jython/background_subtract_batch.jy]: reformulate as above

File: practical/jython/background_subtract_batch.jy [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+# of Fiji (File > New > Script). 
+# Select Python as language in the Scripting Dialog.
+# Error handling is omitted to ease the reading of the script but 
+this should be added
```
Comment (pwalczysko) [inline practical/jython/background_subtract_batch.jy]: strange, this line is not commented out ?

File: practical/jython/background_subtract_batch.jy [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+# Create a dataset to store the newly created images will be added
+name = "script_editor_output_from_dataset_" + dataset_id
+d = DatasetData()
+d.setName(name)
+dm = gateway.getFacility(DataManagerFacility)
+user = gateway.getLoggedInUser()
+ctx = SecurityContext(user.getGroupId())
+d = dm.createDataset(ctx, d, None)
+
+# Loop though each image
```
Comment (pwalczysko) [inline practical/jython/background_subtract_batch.jy]: though -> through

File: practical/python/simple_frap.py (hunk 2 above)
Comment (will-moore) [reply practical/python/simple_frap.py]: Either option sounds good. Probably ```python/server/*``` is good for browsing scripts.

File: practical/jython/crop_rectangle.jy [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+#------------------------------------------------------------------------------
+
+# This Jython script uses ImageJ to Subtract Background.
+# The purpose of the script is to be used in the Scripting Dialog
+# of Fiji (File > New > Script). 
+# Select Python as language in the Scripting Dialog.
+# Error handling is omitted to ease the reading of the script but 
+this should be added
```
Comment (pwalczysko) [inline practical/jython/crop_rectangle.jy]: comment out ?

File: practical/jython/crop_rectangle.jy [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+#  GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+#------------------------------------------------------------------------------
+
+# This Jython script uses ImageJ to Subtract Background.
+# The purpose of the script is to be used in the Scripting Dialog
```
Comment (pwalczysko) [inline practical/jython/crop_rectangle.jy]: Reformulate as above

File: practical/jython/load_thumbnails.jy [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# ------------------------------------------------------------------------------
+
+# This script loads the thumbnails associated to images and open them in
+# ImageJ. The thumbnails are turned into greyscale images in order to be
+# viewed in ImageJ
+# The purpose of the script is to be used in the Scripting Dialog
```
Comment (pwalczysko) [inline practical/jython/load_thumbnails.jy]: reformulate as above

File: practical/jython/load_thumbnails.jy [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+# of Fiji (File > New > Script). 
+# Select Python as language in the Scripting Dialog.
+# Error handling is omitted to ease the reading of the script but
+# this should be added
```
Comment (pwalczysko) [inline practical/jython/load_thumbnails.jy]: here this line is commented out, which signals to me that the two other occassions mentioned above are bugs

File: practical/jython/omero_rois_to_imagej_roi.jy [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# ------------------------------------------------------------------------------
+
+# This script loads the ROIs saved into OMERO and convert them into ImageJ ROIs
+# The rois are then added to the roi manager.
+# The purpose of the script is to be used in the Scripting Dialog
```
Comment (pwalczysko) [inline practical/jython/omero_rois_to_imagej_roi.jy]: reformulate as above

File: practical/jython/load_thumbnails.jy [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# ------------------------------------------------------------------------------
+
+# This script loads the thumbnails associated to images and open them in
```
Comment (pwalczysko) [inline practical/jython/load_thumbnails.jy]: open them -> opens them

File: practical/jython/open_image.jy [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+#  GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# ------------------------------------------------------------------------------
+
+# This script downloads a file an open it in ImageJ using Bio-Formats exporter.
+# The purpose of the script is to be used in the Scripting Dialog
```
Comment (pwalczysko) [inline practical/jython/open_image.jy]: reformulate as above

File: practical/jython/open_image.jy [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# ------------------------------------------------------------------------------
+
+# This script downloads a file an open it in ImageJ using Bio-Formats exporter.
```
Comment (pwalczysko) [inline practical/jython/open_image.jy]: an  open it -> and opens it

File: practical/javascript/figure_labels_from_mapannotation.js (hunk 3 above)
Comment (jburel) [reply practical/javascript/figure_labels_from_mapannotation.js]: That is the original documentation

@will-moore: do you want to suggest something else?

Comment (will-moore) [reply practical/javascript/figure_labels_from_mapannotation.js]: How about this? (to be added before the ```To use...``` line)
// Using the browser devtools to manipulate OMERO.figure is an experimental feature for developers.
// NB: Never paste untrusted code into your browser console.

Comment (pwalczysko) [reply practical/javascript/figure_labels_from_mapannotation.js]: Sounds great.

**Comment by pwalczysko** [comment]
All fixed. Ready to merge FMPOV

**Pull request merged**